### DISCUSSION TITLE
🎉 cibuildwheel is joining the PyPA!

### DISCUSSION THREAD

**Discussion by joerick** [pypa/cibuildwheel#681] [General]
Hello all. Exciting news - cibuildwheel was just accepted into the PyPA!

To quote @pradyunsg on the pypa members list:

> Given that there's no -1s in this vote, we have a clear decision!
> 
> Welcome cibuildwheel to the PyPA! ^.^

[link](https://mail.python.org/archives/list/pypa-committers@python.org/message/ENLRTKKDQYA5ZNCTEI324EAN5MLO6E73/)

I think this is great news for cibuildwheel, and should hopefully set the project up as a stable resource for the community for years to come.

Not a lot is going to change- myself, @YannickJadoul, @mayeut and @henryiii will continue to maintain the project, but there are a few things to be aware of:

- We will be adopting the PyPA code of conduct, which has the same rules as the PSF. See #679.
- The project's URL on Github will change to move under the pypa org. So we'll be pypa/cibuildwheel. Github should take care of redirecting you, so links will continue to work. The name on PyPI will not change. 
- The Github Action name will change to pypa/cibuildwheel. The old `joerick/cibuildwheel` name will continue to work, so this will not require a code change, but it would be a good idea to update your Github Actions YAML when you get a chance :)

🚀

**Comment by joerick** [comment]
In terms of maintainer jobs to support the transition:

- [x] Adopt the PyPA CoC https://github.com/joerick/cibuildwheel/pull/679
- [x] Wait to hear back from Github about the action references
- [x] Initiate the transfer to PyPA
- [ ] Something to think about - while we're at it, changing the default branch from `master` to `main`
- [ ] find/replace in the docs and examples from `joerick/cibuildwheel` to `pypa/cibuildwheel` #682
- [ ] Point the CI providers at the new repo
  - [x] TravisCI
  - [x] Appveyor
  - [x] ReadTheDocs
  - [x] Azure Pipelines
  - [x] CircleCI
- [ ] Perhaps a final 1.x release with these docs updates?

**Reply by henryiii** [reply]
Was going to ask about `master` -> `main`; might be best to wait just a little so we can enjoy the redirect?

If GitHub Actions does not follow redirects (GitHub pages does not), another idea could be to mirror pypa/cibuildwheel as joerick/cibuildhwheel. Sadly, you immediately lose the redirects, but you can keep the old actions working. (Of course, ideally, actions redirect on name changes too!)

**Reply by joerick** [reply]
Here's the response from Github Support.

> Thank you for writing to GitHub Support.
> 
> > Can you confirm if the action will keep working for those users after the move? I'm hoping that it will redirect itself, but just wanted to make sure.
> 
> I can confirm that the redirect will be created so that the action won't break for people that use it in their workflow file.
> 
> But after chatting with the Actions team, their general recommendation would be to leave the original repository where it is, maybe add a note to your repository README that it's deprecated and they should instead use the newly named action repository, and then create a new action repository.
> 
> By leaving the original, things won't accidentally break in the future if your account creates a repository named cibuildwheel which would break the redirect. Maybe unlikely, but could save potential issues in the future.
> 
> I hope this helps. Let me know if you have any further questions.

So the redirect will be made (also corroborated by a few people in the [Python Discourse](https://discuss.python.org/t/cibuildwheel-and-the-pypa/8592/19)). I'm thinking I won't take their suggestion to leave a dummy repo at joerick/cibuildwheel, as I'd rather keep the redirects for links, issues etc.

**Reply by pradyunsg** [reply]
One fun caveat with that, is if you ever fork from pypa/cibuildwheel, the redirects will stop working because your fork will have the same name as the redirects.

**Reply by joerick** [reply]
haha hmm.... forbidden button 😬 thanks for the heads up!

**Reply by YannickJadoul** [reply]
Alternatively, if you want your own fork, @joerick, there might be a way to set up an automatic update of the `master` branch and the tags? If so, then this should be equivalent to the redirect, I would expect. But maybe it's not worth it, if you don't mind committing to pypa/cibuildwheel instead of joerick/cibuildwheel.

**Reply by henryiii** [reply]
That's what I meant by "mirror pypa/cibuildwheel as joerick/cibuildhwheel" and @joerick meant by "dummy repo at joerick/cibuildwheel", I believe; but I think it's much better to just use the redirect.

**Reply by pradyunsg** [reply]
If @joerick wants to keep a fork, it might be a good idea to ask Github support what they can do to help this transfer. 

- Make a temporary GH org.
- Transfer the project to that org.
- Fork that.
- Rename the fork.
- Transfer back to @joerick, under the original name.
- (does @joerick have fork _and_ the original repo?)

**Reply by henryiii** [reply]
If `joerick/cibuildwheel` exists, then `uses: joerick/cibiuldwheel@v1.11.0` will always come from that repo. Now if he accidentally make a fork, then he could delete it, transfer `pypa/cibuildwheel` to `joerick/cibuildwheel` then back to `pypa/cibuildwheel` and that should rebuild the redirect (would be nice if it was configurable). Since joerick can't have a fork of `joerick/cibuildwheel` already (it would be the same repo), then doesn't seem like it's that much of a problem. I'm guessing the plan above would make a differently named fork, like `joerick/cibuildwheel-fork`? I could just make a fork in a different org, rename it, then transfer it to him. But I don't really know if he even wants one. :)

**Reply by joerick** [reply]
Forking to a different org and transferring to my personal account under a different name makes the most sense, I think. But I'm happy to continue working on the pypa repo for now anyway :)

**Reply by YannickJadoul** [reply]
Alright, nevermind too much then; it was just a thought. Sorry for over-complicating things ;-)

**Reply by joerick** [reply]
> Alright, nevermind too much then; it was just a thought. 

All good! Worth having the conversation before I break hundreds of CI setups by pressing a button :)

So! I'll request the transfer now.

**Reply by pradyunsg** [reply]
BTW, the fork back-and-forth I mentioned won't work. From https://docs.github.com/en/github/administering-a-repository/managing-repository-settings/transferring-a-repository:

> The target account must not have a repository with the same name, or a fork in the same network.

**Reply by YannickJadoul** [reply]
Oh, wow, even `git fetch`ing with the old `url = git@github.com:joerick/cibuildwheel.git` in `.git/config` seems to work perfectly!

**Reply by henryiii** [reply]
Yep, GitHub pages is just about the only loss (and we aren't using it). Though I think dependabot may not do the redirect; I had a package on 1.10 wait to ask to update to 1.11 until I had renamed it to pypa/cibuildwheel. But could be wrong, as have no idea why it was not updating to 1.11 before the move :)

**Comment by henryiii** [comment]
The vote was

| +1's | -1's |
|---|---|
| 21 | 0 |

:)

**Reply by henryiii** [reply]
This was one of the most enthusiastic votes I've seen (though we might immediately lose that title to pipx, which just went up for votes).

**Reply by pradyunsg** [reply]
One nit pick -- the votes don't have a `+/-0` option; only +1 or -1. :)
